### PR TITLE
feat: manage presets in upper-bar

### PR DESCRIPTION
**Pull request by lianbenjamin**
### Description of the Changes

manage presets in upper-bar-manager.
each preset should have a `displayedBarWrapper` with icons.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/services/upper-bar-manager/upper-bar-manager.tsx [hunk 1]
```diff
@@ -6,30 +6,37 @@ import { DisplayedBar } from './ui/displayed-bar/displayed-bar.component';
 import { KalturaPluginNames } from '../../types/ui-managers-config';
 const { ReservedPresetAreas, ReservedPresetNames } = ui;
 
+const relevantPresets = Object.values(ReservedPresetNames).filter(
```
Comment (JonathanTGold) [inline src/services/upper-bar-manager/upper-bar-manager.tsx:9]: find more informative name

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

File: src/services/upper-bar-manager/upper-bar-manager.tsx [hunk 2]
```diff
@@ -6,30 +6,37 @@ import { DisplayedBar } from './ui/displayed-bar/displayed-bar.component';
...
     this.injectDisplayedBarComponentWrapper(config.pluginsIconsOrder);
   }
 
   public add(icon: IconDto): number | undefined {
     if (UpperBarManager.validateItem(icon)) {
       const newIcon: IconModel = new IconModel(icon);
       this.componentsRegistry.set(newIcon.id, newIcon);
-      this.displayedBarComponentRef.current!.update();
+      newIcon.presets.forEach((preset) => {
+        this.displayedBarComponentRefs[preset].current?.update();
```
Comment (JonathanTGold) [inline src/services/upper-bar-manager/upper-bar-manager.tsx:38]: the curly braces are redundant, it can formatted to one line

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

File: src/services/upper-bar-manager/upper-bar-manager.tsx [hunk 3]
```diff
@@ -41,7 +48,9 @@ export class UpperBarManager {
     const icon: IconModel | undefined = this.componentsRegistry.get(itemId);
     if (icon) {
       this.componentsRegistry.delete(itemId);
-      this.displayedBarComponentRef.current!.update();
+      icon.presets.forEach((preset) => {
```
Comment (JonathanTGold) [inline src/services/upper-bar-manager/upper-bar-manager.tsx:51]: the curly braces are redundant, it can formatted to one line

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

File: src/services/upper-bar-manager/upper-bar-manager.tsx (hunk 1 above)
Comment (lianbenjamin) [reply src/services/upper-bar-manager/upper-bar-manager.tsx:9]: how about just presets? :)

File: src/services/upper-bar-manager/upper-bar-manager.tsx [hunk 4]
```diff
@@ -6,30 +6,35 @@ import { DisplayedBar } from './ui/displayed-bar/displayed-bar.component';
 import { KalturaPluginNames } from '../../types/ui-managers-config';
 const { ReservedPresetAreas, ReservedPresetNames } = ui;
 
+const presets = Object.values(ReservedPresetNames).filter(
```
Comment (JonathanTGold) [inline src/services/upper-bar-manager/upper-bar-manager.tsx:9]: const should be uppercase by convention and find more descriptive name for the filtered presets

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

**Review by MosheMaorKaltura — APPROVED** [review]

**Pull request merged**